src/common/hooks/useSupportsInterface.ts

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState } from "react";
+import { useContractFunctionHook } from "@govtechsg/ethers-contract-hook";
+import { Contract } from "ethers";
+
+interface Erc165Contract extends Contract {
+  supportsInterface: (interfaceId: []) => Promise<boolean> | undefined;
+}
+
+export const useSupportsInterface = (contractInstance: Erc165Contract | undefined, interfaceId: string) => {
+  const [isInterfaceType, setIsInterfaceType] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+  const [errorMessage, setErrorMessage] = useState<string>();
+
+  const {
+    call: supportsInterface,
+    value: isSameInterfaceType,
+    errorMessage: supportInterfaceErrorMessage,
+    state,
+  } = useContractFunctionHook(contractInstance, "supportsInterface");
+
+  useEffect(() => {
+    setIsLoading(true);
+    supportsInterface(interfaceId);
+  }, [interfaceId, supportsInterface]);
+
+  useEffect(() => {
+    if (state === "ERROR") {
```

Comment:
not sure if there's an exported type for state, if there is can use that instead of a hardcoded string

Reply:
There is an exported type under `ContractFunctionState`, but this uses string literals instead of enums, not sure if its possible to use the type as a value.

Reply:
Don't think there is a way to convert string literal type to enum unless I declare an enum in the hook. But writing something else in the === throws a type error so... whats best practice here?

Reply:
just leave it as it is then